resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/stream/config/AkkaRunnableGraph.java

```diff
@@ -67,69 +70,73 @@ KafkaProducerService kafkaProducerService(final Sink<ProducerRecord<String, Reso
 
     @Bean
     Function1<Throwable, Directive> supervisor() {
-        return ex -> Supervision.resumingDecider().apply(ex);
+        return (Throwable ex) -> {
+            LOGGER.error("Fatal error during stream processing, element will be dropped: ", ex);
+            return Supervision.resumingDecider().apply(ex);
+        };
     }
 
     @Bean
     RunnableGraph<DrainingControl<Done>> runner(
             final Source<CommittableMessage<String, ResourceRequest>, Control> source,
-            final Sink<Envelope<String, ResourceResponse, Committable>, CompletionStage<Done>> sink,
+            final Sink<Envelope<String, byte[], Committable>, CompletionStage<Done>> sink,
             final Function1<Throwable, Directive> supervisionStrategy,
             final ProducerTopicConfiguration topicConfiguration,
-            final StreamingResourceServiceProxy service) {
+            final ResourceServicePersistenceProxy service) {
         // Get output topic from config
         Topic outputTopic = topicConfiguration.getTopics().get("output-topic");
+        // Get error topic from config
+        Topic errorTopic = topicConfiguration.getTopics().get("error-topic");
 
         // Read messages from the stream source
         return source
                 .map(message -> new Pair<>(message, Optional.ofNullable(message.record().value())))
-                // Get a stream of resources from an iterator
+
+                // Get the request optional from the Pair
                 .flatMapConcat(messageAndRequest -> messageAndRequest.second()
                         // If the request optional is not empty then we need to process the message
                         // Get a stream of resources from the implemented resource-service
-                        .map(request -> Source.fromIterator(() -> service.getResourcesById(request.resourceId))
+                        .map(request -> service.getResourcesById(request)
```

Comment:
Think there is an earlier comment that we could possibly rename this method?

Reply:
Please see above comment.